nifi-frontend/src/main/frontend/apps/nifi-registry/src/app/pages/buckets/feature/ui/bucket-table-filter/bucket-table-filter.component.ts

```diff
@@ -0,0 +1,103 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, EventEmitter, Input, Output, inject } from '@angular/core';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatInputModule } from '@angular/material/input';
+import { MatSelectModule } from '@angular/material/select';
+import { FormBuilder, FormGroup, ReactiveFormsModule } from '@angular/forms';
+import { debounceTime } from 'rxjs';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+
+export interface BucketTableFilterColumn {
+    key: string;
+    label: string;
+}
+
+export interface BucketTableFilterContext {
+    filterTerm: string;
+    filterColumn: string;
+    changedField: string;
+}
+
+@Component({
+    selector: 'bucket-table-filter',
+    templateUrl: './bucket-table-filter.component.html',
+    styleUrl: './bucket-table-filter.component.scss',
+    standalone: true,
+    imports: [ReactiveFormsModule, MatFormFieldModule, MatInputModule, MatSelectModule]
+})
+export class BucketTableFilterComponent {
+    private formBuilder = inject(FormBuilder);
+
+    filterForm: FormGroup = this.formBuilder.group({
+        filterTerm: '',
+        filterColumn: ''
+    });
+
+    private _filterableColumns: BucketTableFilterColumn[] = [];
+
+    @Input() set filterableColumns(columns: BucketTableFilterColumn[]) {
+        this._filterableColumns = columns ?? [];
+        if (this._filterableColumns.length > 0) {
+            const current = this.filterForm.get('filterColumn')?.value;
+            const valid = this._filterableColumns.some((column) => column.key === current);
+            const valueToApply = valid ? current : this._filterableColumns[0].key;
+            this.filterForm.get('filterColumn')?.setValue(valueToApply, { emitEvent: false });
+        }
+    }
+    get filterableColumns(): BucketTableFilterColumn[] {
+        return this._filterableColumns;
+    }
+
+    @Input() set filterTerm(term: string) {
+        this.filterForm.get('filterTerm')?.setValue(term ?? '', { emitEvent: false });
+    }
+
+    @Input() set filterColumn(column: string) {
+        if (column) {
+            this.filterForm.get('filterColumn')?.setValue(column, { emitEvent: false });
+        } else if (this._filterableColumns.length > 0) {
+            this.filterForm.get('filterColumn')?.setValue(this._filterableColumns[0].key, { emitEvent: false });
+        }
+    }
+
+    @Input() filteredCount = 0;
+    @Input() totalCount = 0;
+
+    @Output() filterChanged: EventEmitter<BucketTableFilterContext> = new EventEmitter<BucketTableFilterContext>();
+
+    constructor() {
+        this.filterForm
+            .get('filterTerm')
+            ?.valueChanges.pipe(debounceTime(250), takeUntilDestroyed())
```

Comment:
I used a debounceTime of 500ms over in the flows table. I have no problem using 250ms or whatever value NiFi uses, but maybe we should store this in an environment variable to ensure consistency across the application.